### ISSUE TITLE
After upgrade to the second release candidate, I lose the spyder-notebook plugin.

### ISSUE THREAD

**Issue by lwbaqueros** [spyder-ide/spyder#10824]
### What steps will reproduce the problem?

I upgrade from a previous reeleasee spyder beta to the last candidate. 

Then the spyder-notebook was disable and did not work anymore. 

when I do conda list, I can see the package there, I uninstall the package and installa againg, but it did not work. 

Then I download and copy the plugin in to the folder home/user/.spyder/plugin and then it appears, but it doesn't work

### Traceback
```python-traceback
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\http\client.py", line 1252, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\http\client.py", line 1298, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\http\client.py", line 1247, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\http\client.py", line 1026, in _send_output
    self.send(msg)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\http\client.py", line 966, in send
    self.connect()
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\connection.py", line 181, in connect
    conn = self._new_conn()
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\connection.py", line 168, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x00000290D1A5A488>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\adapters.py", line 449, in send
    timeout=timeout
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\urllib3\util\retry.py", line 399, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8889): Max retries exceeded with url: /api/sessions?token=7d425e364577ecaf1405a455477b45da99a155fe314775cb (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x00000290D1A5A488>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Lucho Baquero\.spyder-py3-dev\plugins\spyder_notebook\notebookplugin.py", line 377, in close_client
    client.shutdown_kernel()
  File "C:\Users\Lucho Baquero\.spyder-py3-dev\plugins\spyder_notebook\widgets\client.py", line 248, in shutdown_kernel
    kernel_id = self.get_kernel_id()
  File "C:\Users\Lucho Baquero\.spyder-py3-dev\plugins\spyder_notebook\widgets\client.py", line 232, in get_kernel_id
    sessions_req = requests.get(sessions_url).content.decode()
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "F:\Users\lbaquero\Anaconda3\envs\spyder-beta\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8889): Max retries exceeded with url: /api/sessions?token=7d425e364577ecaf1405a455477b45da99a155fe314775cb (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x00000290D1A5A488>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
```

## Versions

* Spyder version: 4.0.0rc2 
* Python version: 3.7.5
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.4.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.5 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.0;<2.0.0:  1.8.0 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.9.0 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
pandas >=0.13.1              :  0.24.2 (OK)
numpy >=1.7                  :  1.17.3 (OK)
scipy >=0.17.0               :  None (NOK)
pyls >=0.31.0;<0.32.0        :  0.31.1 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

**Comment by lwbaqueros** [comment]
I installed in a new environment with spyder= **4.0.0b2** version, and then installed the spyder-notebook  pluging = **it works perfectly.** 

Then, I repeated with another new environment + spyder=**4.0.0rc2** + spyder-notebook installation  = **It doesn't Work**

In the release candidate 2, it seems this plugin doesn't work. I tried to install it manually, and somehow it could appear in the spyder panes, show the tab, but when you want to open a Jupyter notebook it crushed giving many errors. 

Could you please guy check if this plugin works for you in 4.0.0rc2... and give any advice.... I really want to use the second candidate because of KITE

**Comment by jitseniesen** [comment]
You are right, the notebook plugin (and all other plugins) do not work with the release candidate versions of Spyder because of some changes in the plugin API. We have made the necessary changes to the notebook plugin and some other plugins but not yet released new versions. If you want to run spyder 4.0.0rc2 and the notebook plugin, then you have to install the development version of the notebook plugin.

**Comment by jitseniesen** [comment]
For the record, this was fixed in spyder-ide/spyder-notebook#218 .
